packages/core/test/SchemaField_test.js

```diff
@@ -292,6 +292,36 @@ describe("SchemaField", () => {
     });
   });
 
+  describe("additional properties", () => {
+    const schema = {
+      type: "object",
+      additionalProperties: true,
+    };
+
+    it("should render text input for object key if schema has additional properties", () => {
+      const { node } = createFormComponent({ schema });
+      Simulate.click(node.querySelector(".glyphicon-plus"));
+
+      expect(node.querySelectorAll("input[value='newKey']")).to.have.length.of(
+        1
+      );
+    });
+
+    it("should render SchemaField as input for object key if schema has propertyNames schema", () => {
+      const schemaWithPropertyNames = {
+        ...schema,
+        propertyNames: {
+          type: "string",
+          enum: ["1", "2"],
```

Comment:
Can you add a test to see if you just set propertyNames equal to something like `"pattern": "^[A-Za-z_][A-Za-z0-9_]*$"` (without specifying a `type` explicitly), it still works?

Reply:
Sounds good, I'll add them!